src/lib/autoinstall/xml_validator.rb

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+Yast.import "XML"
+
+module Y2Autoinstallation
+  # Validates an XML document against a given RNG schema
+  class XmlValidator
+    include Yast::Logger
+
+    attr_reader :xml, :schema
+
+    # Constructor
+    # @param xml [String] XML document or path to a XML file
```

Comment:
Really a whole document, which will be logged  by `#validate` below?

Reply:
Ah, good point with logging whole document...

Reply:
BTW that's how [Yast::XML.validate](https://github.com/yast/yast-yast2/blob/20f7696e2836cf93647d925c8ce612a8c2771666/library/xml/src/modules/XML.rb#L202) works.... Yes, a bit strange...

Reply:
Fixed in https://github.com/yast/yast-yast2/pull/1064